specs/interop.md

```diff
@@ -0,0 +1,188 @@
+# Interoperability
+
+The ability for a blockchain to easily read the state of another blockchain is called interoperability.
+Low latency interoperability allows for horizontally scalable blockchains, a key feature of the superchain.
+
+Note: this document references an "interop network upgrade" as a temporary name. The actual name of the
+network upgrade will be included in this document in the future.
+
+## Overview
+
+| Term          | Definition                                               |
+|---------------|----------------------------------------------------------|
+| Local Chain   | A blockchain that includes an initiating message |
+| Remote Chain  | A blockchain that includes a relaying message |
+| Initiating Message | A transaction submitted to a local chain that authorizes execution on a remote chain |
+| Relaying Message | A transaction submitted to a remote chain that corresponds to an initiating message |
+| Cross Chain Message | The cumulative execution and side effects of the initiating message and relaying message |
+| Interop Set | The set of chains that originate initiating transactions where the relaying transactions are valid |
+
+A total of two transactions are required to complete a cross chain message. The first transaction is submitted
+to the local chain which authorizes execution on a remote chain. The second transaction is submitted to a
+remote chain, where the block builder SHOULD only include it if they are certain that the first transaction was
+included in the local chain.
+
+The term "block builder" is used interchangeably with the term "sequencer" for the purposes of this document but they need not be the same
+entity in practice.
+
+## The Interop Set
+
+The interop set defines the set of chains that a remote chains allows as local chains. Another way of
+saying it is that the interop set defines the set of initiating messages that are valid for a relaying
+message to be included. A relaying message MUST have an initiating message that is included in a chain
+in the interop set.
+
+The interop set is defined by chain id. Since it is impossible to enforce uniqueness of chain ids,
```

Comment:
Does it make sense to hold reference of the system_config_address which maps to a chain id in the interop state?

That way even though chain ids are not unique, it can provable

Reply:
how does social consensus work with fault proofs? The superchain registry is off-chain as of now right?

Reply:
Hmm I don't understand what you mean by this. No matter what, there is no canonical mapping between a chain id and rpc url. Social consensus is used to determine the "type" of rpc url aka what chain it points to. Social consensus tracks the canonical chain that uses a chain id, just like how we use https://chainlist.org/ today

Reply:
Since the L1 address space is shared between chains, my question was if there's any value in using it so that we canonical label for a chain that's verifiable.

I'm not super familiar with fault proofs and was wondering how the right preimage inputs would be provided to a proof on L1 when a chain-dependency input is invalid if the chain id plays into that (if it doesn't then that answers my q)

Reply:
We don't necessarily want to put labels on L1 and it also doesn't solve the problem because a label is just an alias for a chainid, who decides what the canonical labels are for chainids? No matter what there is a subjective nature to resolving a chainid to a particular chain where people agree on what chain id maps to what chain via social consensus